tests/kernel/gen_isr_table/src/main.c

```diff
@@ -32,6 +32,14 @@ extern u32_t _irq_vector_table[];
 #define ISR5_OFFSET	4
 #define ISR6_OFFSET	5
 
+#if defined(CONFIG_SOC_ARC_EMSDP)
```

Comment:
I would do this workaround in the testcase.yaml instead of here. Lets keep the test soc code free.

Reply:
ok, will do that

Reply:
I think we can't  change  CONFIG_NUM_IRQS in testcase.yaml.  For emsdp's case , the console irq will use irq 107/108,  if change CONFIG_NUM_IRQS to 105 globally,  console's irq cannot be used , then no console out.


A fast fix is just like what i did, although it's ugly.

A better fix is to let IRQ_LINE not rely on CONFIG_NUM_IRQS,  but a value may be redefined by application or boards, e.g,  CONFIG_TEST_NUM_IRQS whose default val is CONFIG_NUM_IRQS.